lib/path.js

```diff
@@ -39,16 +45,30 @@ const isWsp = (c) => {
   );
 };
 
+/**
+ * @param {string} c
+ */
 const isDigit = (c) => {
   const codePoint = c.codePointAt(0);
+  if (codePoint == null) {
+    return false;
+  }
   return 48 <= codePoint && codePoint <= 57;
 };
 
+/**
+ * @typedef {'none' | 'sign' | 'whole' | 'decimal_point' | 'decimal' | 'e' | 'exponent_sign' | 'exponent'} ReadNumberState
+ */
+
+/**
+ * @param {string} string
+ * @param {number} cursor
+ * @return {[number, number | null]}
+ */
 const readNumber = (string, cursor) => {
   let i = cursor;
   let value = '';
-  // none | sign | whole | decimal_point | decimal | e | exponent_sign | exponent
-  let state = 'none';
+  let state = /** @type {ReadNumberState} */ ('none');
```

Comment:
Did prettier wrap 'none' with parentheses?

Reply:
Casts in JSDoc require the parentheses.